### PR TITLE
Change retries in client to have exponential sleep time.

### LINKED ISSUE #411: Exponential retries in client
The client supports retrying requests for predictions, but it does the requests right after another. Maybe better to do exponential backof? I found the python library [tenacity](https://tenacity.readthedocs.io/en/latest/) which is supposed to make retries easy and nice, and it includes exponential retries as one method. Maybe use it instead of our own little loop? And then maybe also do #343 at the same time?

### PR DESCRIPTION
**Pull request by EinarJohnsen**
Change from the default 5 seconds sleep to exponential sleep based on number of retries. Fixes #411

### CODE REVIEW

File: tests/gordo_components/client/test_client.py [hunk 1]
```diff
@@ -490,3 +493,29 @@ def test_client_endpoint_filtering(
...
+
+def test_exponential_sleep_time(watchman_service):
+    endpoint = _endpoint_metadata("t1", False)
+
+    start, end = (
+        isoparse("2016-01-01T00:00:00+00:00"),
+        isoparse("2016-01-01T12:00:00+00:00"),
+    )
+
+    with patch(
```
Comment (epa095) [inline tests/gordo_components/client/test_client.py:506]: nice

File: tests/gordo_components/client/test_client.py [hunk 2]
```diff
@@ -490,3 +493,29 @@ def test_client_endpoint_filtering(
...
+        y = None
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(
+            client._process_post_prediction_task(
+                X=x, y=y, chunk=slice(0, 1), endpoint=endpoint, start=start, end=end
+            )
+        )
+        loop.close()
+
+    calls = [call(8), call(16), call(32), call(64), call(128)]
```
Comment (epa095) [inline tests/gordo_components/client/test_client.py]: maybe call this variable something slightly more expressive, like "expected_calls" or something?

**Review by epa095 — COMMENTED** [review]
Looking good. But maybe mention the exponential retries in the docstring of the class?

File: tests/gordo_components/client/test_client.py [hunk 3]
```diff
@@ -490,3 +493,32 @@ def test_client_endpoint_filtering(
...
+                chunk=slice(0, 1),
+                endpoint=endpoint,
+                start=start,
+                end=end,
+            )
+        )
+        loop.close()
+
+    expected_calls = [call(8), call(16), call(32), call(64), call(128)]
+    time_sleep.assert_has_calls(expected_calls)
```
Comment (milesgranger) [inline tests/gordo_components/client/test_client.py:524]: Cool! :mountain_cableway:

**Review by milesgranger — APPROVED** [review]
Looks good to me, sir! :bowtie:

**Pull request merged**